src/main/java/org/neo4j/gis/spatial/rtree/RTreeIndex.java

```diff
@@ -186,11 +187,12 @@ public void add(List<Node> geomNodes) {
 				nodesToAdd.add(n);
 			}
 			nodesToAdd.addAll(geomNodes);
-			for (Node n : getAllIndexInternalNodes()) {
-				if (!n.equals(getIndexRoot())) {
-					deleteNode(n);
-				}
-			}
+
+			final Set<Node> toDelete = StreamSupport.stream(getAllIndexInternalNodes().spliterator(), false)
+					.filter(node -> !node.equals(getIndexRoot()))
+					.collect(Collectors.toSet());
+			toDelete.stream().forEach(node -> deleteNode(node));
```

Comment:
In 3.2.2 the old code had thrown an error, try to use `Iterator.remove()` didn't work (UnsupportedOperationException or a similar one). Since the iterator approach fails, I came up with this one. Can you think of a smarter solution?